server/pulp/server/webservices/views/serializers/dispatch.py

```diff
@@ -47,3 +48,21 @@ def task_status(task):
     task_dict['id'] = str(task['id'])
 
     return task_dict
+
+
+def group_task_summary(tasks):
+    """
+    Return summary of states for tasks in QuerySet
+
+    :param tasks: QuerySet of tasks
+    :return: dictionary summarizing states of tasks in QuerySet
+    """
+    summary = {'total': tasks.count()}
+
+    for state in CALL_STATES:
+        summary[state] = 0
+
+    for task in tasks:
```

Comment:
It would be valuable to limit which fields are returned, since you only need a small subset of the total task document. Doing so in this function is valuable, in case the calling function wants to use the original queryset for other purposes and needs more fields.